kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.session
+
+import org.apache.hive.service.rpc.thrift.TProtocolVersion
+
+import org.apache.kyuubi.events.{EventBus, KyuubiEvent, KyuubiSessionEvent}
+
+abstract class KyuubiSession(
+    protocol: TProtocolVersion,
+    user: String,
+    password: String,
+    ipAddress: String,
+    conf: Map[String, String],
+    sessionManager: KyuubiSessionManager)
+  extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
+
+  protected val sessionEvent = KyuubiSessionEvent(this)
+  EventBus.post(sessionEvent)
+
+  def getSessionEvent: Option[KyuubiEvent] = {
```

Comment:
shall we just return `Option[KyuubiSessionEvent]` ?

Reply:
do we define only an empty getSession method in KyuubiSession class?

Reply:
thanks, done